src/announcements.ts

```diff
@@ -0,0 +1,65 @@
+import { Location } from 'swup';
```

Comment:
This is a lot of changes. I guess due to already happening refactoring? Well, I'll have to rely on your manual tests for this anyways as I'm too busy to test it myself right now.

Reply:
Yeah, it would have been a bit silly not to refactor here as it made everything twice as simple to change. I'm definitely planning on a) doing another final round of manual testing and b) writing end-to-end tests before releasing anything.